### ISSUE TITLE
Chapter 3 replication failure

### ISSUE THREAD

**Issue by khgitting** [CamDavidsonPilon/Probabilistic-Programming-and-Bayesian-Methods-for-Hackers#245]
Hi there--even after checking out a clean Chapter3.ipynb from master, the model consistently fails to learn appropriate cluster0 std and p (frequency of assignment to cluster0).

Expected behavior, 1:
![image](https://cloud.githubusercontent.com/assets/1838992/5894541/5751811e-a4d7-11e4-8890-309b2dabd0ee.png)

Actual behavior, 1:
![image](https://cloud.githubusercontent.com/assets/1838992/5894545/65f34b6c-a4d7-11e4-92c4-bddd31177151.png)

Expected behavior, 2:
![image](https://cloud.githubusercontent.com/assets/1838992/5894590/95a1544e-a4d7-11e4-9041-729ccf869697.png)

Actual behavior, 2:
![image](https://cloud.githubusercontent.com/assets/1838992/5894592/a352c244-a4d7-11e4-9c0a-4af9dcc0a1a8.png)

Any idea what could be causing that?

**Comment by seriousblack** [comment]
Hmmm….

This is interesting...

**Comment by CamDavidsonPilon** [comment]
hey @khgitting, I've noticed this too as well - I can replicate your behaviour as well. To confirm, can you post your version of PyMC? 

```
import pymc as mc
mc.__version__
```

**Comment by khgitting** [comment]
I'm on version 2.3.4

**Comment by erdman** [comment]
I also failed to replicate with PyMC version 2.3.4 (with same results as charted above).  Reverting PyMC to version 2.2 works as intended.

**Comment by khgitting** [comment]
You can also get sensible results (still with 2.3.4) by switching the prior of p (probability of assignment to cluster 1) from a noninformative Uniform prior to a Beta(x,x) for x >= 20 or so, representing some belief (the higher x, the stronger that belief) that p is somewhat near .5.

**Comment by vumaasha** [comment]
+1 I also face this problem.

**Comment by QuentinAndre** [comment]
+1, Same issue here. Python 3.4, PyMC version 2.3.4.

It would be interesting to know what has chaned from 2.2 to 2.3.4, but I'm not well versed enough into PyMC to figure this out on my own...

**Comment by EDiLD** [comment]
:+1:  Same here. Problems with both pymc2 and pymc3 :(

**Comment by clarknie** [comment]
+1 for the same issue.

**Comment by clarknie** [comment]
Yes, getting back to pymc @2.2 would solve the issue as @erdman suggested. But we want to know use the new version for other improvements, don't we @QuentinAndre? 

I submitted a issue in pymc in the hope of getting some fix soon:
https://github.com/pymc-devs/pymc/issues/46

**Comment by fonnesbeck** [comment]
Does this work with a build of the current master?

**Comment by clarknie** [comment]
@fonnesbeck Yes. I just build of the current master and found the version to be pymc @ 2.3.5. It's the version 2.3.4 that I installed from macPort that is causing the issue.

**Comment by fonnesbeck** [comment]
Ok, then it's probably time for a 2.3.5 bug fix release for PyMC.

**Comment by 18380584875** [comment]
When I run the code which is produced the sample, the python will stop run it. why?

**Comment by 18380584875** [comment]
I find the code (step2 = pm.ElemwiseCategorical(vars=[assignment])) will take place wrong through I checked. but I don't know how to deal with it.
